### PR TITLE
refactor: Don't use skew_ and translate_ attributes on SVGs to animate blocks.

### LINKED ISSUE #6595: Refactor block animations to not store data on SVG elements
**Describe the bug**

Currently the block animations code stores data on properties of SVG element objects, which are also referenced in the main block code. This design has caused us problems in the past (namely https://github.com/google/blockly/issues/6541) because it is difficult to refactor the code when you're dealing with side effects on "private" properties that are triggered by other files.

**Suggested solution**

I think the easiest thing is just to add some public getters and setters for handling the skew and the translation (so that the block code and the block animation code can both mess with it). But there may also be a more elegant solution that does better at information hiding.

**Additional Info**

Related to #6559

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] I ran `npm run format` and `npm run lint`

### Resolves
Fixes #6595

### Proposed Changes
This change stops setting skew_ and translation_ attributes on SVGs for blocks. It introduces an internal getter for the translation, and stores it in a proper field on BlockSvg.

skew_ was only being set to a non-'' value in one place, and then immediately read back to set the transform. Instead, the transform is simply setting using the directly-calculated skew value, so there's no need to carry it around and clear it at other points.

WorkspaceCommentSvg also maintained the skew_ and translate_ attributes, but they appear to be vestigial from a long-ago copy-paste and not actually used, so they were removed.

### CODE REVIEW

File: core/block_svg.ts [hunk 1]
```diff
@@ -506,13 +516,8 @@ export class BlockSvg extends Block implements IASTNodeLocationSvg,
       this.workspace.getBlockDragSurface()!.translateSurface(
           newLoc.x, newLoc.y);
     } else {
-      (this.svgGroup_ as AnyDuringMigration).translate_ =
-          'translate(' + newLoc.x + ',' + newLoc.y + ')';
-      (this.svgGroup_ as AnyDuringMigration)
-          .setAttribute(
-              'transform',
-              (this.svgGroup_ as AnyDuringMigration).translate_ +
-                  (this.svgGroup_ as AnyDuringMigration).skew_);
```
Comment (BeksOmega) [inline core/block_svg.ts:515]: Why did you remove setting the skew value here? Won't this mess up the animation where the block wobbles when you disconnect it?

Or was the assignment in `setDragging` overwriting the skew value anyway, so it doesn't matter?

File: core/block_animations.ts [hunk 2]
```diff
@@ -28,8 +28,8 @@ interface CloneRect {
 /** PID of disconnect UI animation.  There can only be one at a time. */
 let disconnectPid: ReturnType<typeof setTimeout>|null = null;
 
-/** SVG group of wobbling block.  There can only be one at a time. */
-let disconnectGroup: SVGElement|null = null;
+/** The wobbling block.  There can only be one at a time. */
+let wobblingBlock: BlockSvg|null = null;
```
Comment (BeksOmega) [inline core/block_animations.ts:32]: Thank you for the rename this is much better haha.

File: core/block_animations.ts [hunk 3]
```diff
@@ -202,13 +200,12 @@ function disconnectUiStep(group: SVGElement, magnitude: number, start: Date) {
  * @internal
  */
 export function disconnectUiStop() {
-  if (disconnectGroup) {
+  if (wobblingBlock) {
```
Comment (BeksOmega) [inline core/block_animations.ts:203]: Optional nit: `if (!wobblingBlock) return;`

File: core/block_svg.ts (hunk 1 above)
Comment (gonfunko) [reply core/block_svg.ts:515]: AFAICT no, because the disconnect animation calculates skew (and updates `transform` using it) in a 10ms loop, so it will continue to do so and override the lack of skew from here. Experimentally there was no visually perceptible difference and the animation was still shown, but please do double check me on this!

Comment (BeksOmega) [reply core/block_svg.ts:515]: Pulled it down and checked, and I agree looks fine =) Sweetness!

**Review by BeksOmega — APPROVED** [review]

File: core/block_animations.ts (hunk 3 above)
Comment (gonfunko) [reply core/block_animations.ts:203]: Done.
